src/hooks/useComponentVisible.tsx

```diff
@@ -1,13 +1,15 @@
 import { useState, useEffect, useRef } from 'react';
 
-export default function useComponentVisible(initialIsVisible) {
+export default function useComponentVisible(
+	ref: React.RefObject<HTMLElement>,
```

Comment:
I'm still not sure what exactly the ref is, and what value it holds in the ref?

Reply:
@BraidenCutforth

Reply:
@seiyaterada A ref is esentially a pointer to an HTMLElement. In this case it is a pointer to the container which is using it, to detect clicks outside of it. (Or really detect if the click was inside the container...)

Reply:
As an added note for this to make sense, the dom is a tree of elements, and an element has a reference to the parent that is above it.